subscriber_notifee.go

```diff
@@ -14,82 +16,201 @@ import (
 // subscriberNotifee implements network.Notifee and also manages the subscriber to the event bus. We consume peer
 // identification events to trigger inclusion in the routing table, and we consume Disconnected events to eject peers
 // from it.
-type subscriberNotifee IpfsDHT
-
-func (nn *subscriberNotifee) DHT() *IpfsDHT {
-	return (*IpfsDHT)(nn)
+type subscriberNotifee struct {
+	dht                                    *IpfsDHT
+	identifySub, updateSub, routabilitySub event.Subscription
 }
 
-func (nn *subscriberNotifee) subscribe(proc goprocess.Process) {
-	dht := nn.DHT()
-
-	dht.host.Network().Notify(nn)
-	defer dht.host.Network().StopNotify(nn)
+func newSubscriberNotifiee(dht *IpfsDHT) (*subscriberNotifee, error) {
```

Comment:
note: given that

1. The routability sub remembers the last value.
2. We now walk through all connected peers on start.

We don't _have_ to do this synchronously on start.

Reply:
Seems safer to do this on start. Also means that we get errors instead of panics which seems nicer.